---
title: C28105 | Документация Майкрософт
ms.date: 11/15/2016
ms.prod: visual-studio-dev14
ms.technology: vs-ide-code-analysis
ms.topic: reference
f1_keywords:
- C28105
helpviewer_keywords:
- C28105
ms.assetid: 5afdec78-30eb-470c-980f-93d0cb3e74cc
caps.latest.revision: 5
author: mikeblome
ms.author: mblome
manager: jillfra
ms.openlocfilehash: 78fefc8101db0585d4e7fb741dd00ff9473698c8
ms.sourcegitcommit: 94b3a052fb1229c7e7f8804b09c1d403385c7630
ms.translationtype: MT
ms.contentlocale: ru-RU
ms.lasthandoff: 04/23/2019
ms.locfileid: "68156404"
---
# <a name="c28105"></a>C28105
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

Предупреждение C28105: Утечка ресурса вследствие исключения  
  
 Указанный ресурс не освобождается, когда возникает исключение. Инструкция, указанная в пути может вызвать исключение. Это предупреждение аналогично предупреждению [C28103](../code-quality/c28103.md), за исключением того, что в этом случае происходит исключение.  
  
## <a name="example"></a>Пример  
 Следующий пример кода вызывает появление этого предупреждения.  
  
```  
res = KeSaveFloatingPointState(buffer);  
  
res = AllocateResource(Resource);  
char *p2 = new char[10]; // could throw  
  
delete[] p2;  
FreeResource(Resource)  
```  
  
 Следующий пример кода позволяет избежать появления этого предупреждения.  
  
```  
res = AllocateResource(Resource);  
char *p2;  
  
try {  
    p2 = new char[10];  
} catch (std::bad_alloc *e) {  
    // just handle the throw  
    ;  
}  
FreeResource(Resource)  
```
